agent/proxycfg/testing.go

```diff
@@ -137,11 +137,51 @@ func TestUpstreamNodes(t testing.T) structs.CheckServiceNodes {
 				Address:    "10.10.1.2",
 				Datacenter: "dc1",
 			},
-			Service: structs.TestNodeService(t),
+			Service: structs.TestNodeServiceWithName(t, service),
 		},
 	}
 }
 
+func TestPreparedQueryNodes(t testing.T, service string) structs.CheckServiceNodes {
+
+	// The service instances targeted by the prepared query are given the slightly different name
+	// "geo-cache-target" to ensure we don't use the prepared query's name for SAN validation.
+	// The name of prepared queries won't always match the name of the service they target.
```

Comment:
This comment only makes sense in the context of a singular test that passes `"geo-cache"` as `service`. Should we make it more generic and put it at the function level?

Reply:
Good point, just updated that.